tools/pythonpkg/tests/fast/arrow/test_arrow_replacement_scan.py

```diff
@@ -0,0 +1,38 @@
+import duckdb
+import pytest
+import os
+import pandas as pd
+try:
+    import pyarrow.parquet as pq
+    import pyarrow.dataset as ds
+    can_run = True
+except:
+    can_run = False
+
+class TestArrowReplacementScan(object):
+    def test_arrow_table_replacement_scan(self, duckdb_cursor):
+        if not can_run:
+            return
+
+        parquet_filename = os.path.join(os.path.dirname(os.path.realpath(__file__)),'data','userdata1.parquet')
+        userdata_parquet_table = pq.read_table(parquet_filename)
+        con = duckdb.connect()
+        assert con.execute("select count(*) from userdata_parquet_table").fetchone() ==  (1000,)
```

Comment:
Could you add some tests where:
* We re-use the same arrow table/pandas df multiple times in replacement scans
* We create a view using the replacement scan, then delete the arrow table/pandas df, then query the view